test/unit/io/alignment_file/alignment_file_format_test_template.hpp

```diff
@@ -495,6 +495,47 @@ TYPED_TEST_P(alignment_file_write, with_header)
     EXPECT_EQ(this->ostream.str(), this->verbose_output);
 }
 
+TYPED_TEST_P(alignment_file_write, cigar_vector)
+{
+    std::vector<std::vector<cigar>> cigar_v
+    {
+        {{1, 'S'_cigar_op}, {1, 'M'_cigar_op}, {1, 'D'_cigar_op}, {1, 'M'_cigar_op}, {1, 'I'_cigar_op}},
+        {{7, 'M'_cigar_op}, {1, 'D'_cigar_op}, {1, 'M'_cigar_op}, {1, 'S'_cigar_op}},
+        {{1, 'S'_cigar_op}, {1, 'M'_cigar_op}, {1, 'P'_cigar_op}, {1, 'M'_cigar_op}, {1, 'I'_cigar_op},
+         {1, 'M'_cigar_op}, {1, 'I'_cigar_op}, {1, 'D'_cigar_op}, {1, 'M'_cigar_op}, {1, 'S'_cigar_op}}
+    };
+
+    {
+        this->tag_dicts[0]["NM"_tag] = 7;
+        this->tag_dicts[0]["AS"_tag] = 2;
+        this->tag_dicts[1]["xy"_tag] = std::vector<uint16_t>{3,4,5};
+
+        alignment_file_output fout{this->ostream, TypeParam{}, fields<field::HEADER_PTR,
+                                                                      field::ID,
+                                                                      field::FLAG,
+                                                                      field::REF_ID,
+                                                                      field::REF_OFFSET,
+                                                                      field::MAPQ,
+                                                                      field::CIGAR, // cigar instead of alignment
+                                                                      field::OFFSET,
+                                                                      field::MATE,
+                                                                      field::SEQ,
+                                                                      field::QUAL,
+                                                                      field::TAGS>{}};
+
+        for (size_t i = 0; i < 3; ++i)
```

Comment:
Use `0ul` and `3ul` literals because you compare unsigned long values.
(Doesn't it trigger a warning?)

Reply:
nope ^^. I changed it